src/main/java/com/sforce/cd/apexUnit/client/utils/ApexClassFetcherUtils.java

```diff
@@ -321,8 +322,8 @@ public static String fetchAndAddToMapApexClassIdBasedOnName(PartnerConnection co
 		}
 		if (queryResult != null && queryResult.getDone()) {
 			String[] classIds = fetchApexClassesAsArray(queryResult);
-			for (String classId : classIds) {
-				apexClassId = classId;
+			if(classIds.length>0){
```

Comment:
Thanks @pmedapuram 
Null can be returned if queryResult.getDone() is false or if there are no records returned from the query. 
I see your code tested for # of records returned. Did your code return false for queryResult.getDone() ? I dont see queryResult.getDone() in your code.